streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -102,7 +102,7 @@ class Checkpoint(ssc: StreamingContext, val checkpointTime: Time)
 private[streaming]
 object Checkpoint extends Logging {
   val PREFIX = "checkpoint-"
-  val REGEX = (PREFIX + """([\d]+)([\w\.]*)""").r
+  val REGEX = (PREFIX + """([\d]{9,})([\w\.]*)""").r
```

Comment:
This seems to make it always expect 9+ digits, whereas it accepts 1 or more now. Your case is `checkpoint-01` so I'm missing how this works?

Reply:
Please read my previous answer to HyukjinKwon. I explained there what was intended.
In my case `checkpoint-01` is a checkpoint **directory** name which matches positively using current regex accepting 1 or more digits.